python/paddle/fluid/contrib/int8_inference/utility.py

```diff
@@ -423,107 +424,83 @@ def __get_quantize_dequantize_combination(self, program):
-                        op_type = program.current_block().ops[op_index].type
-                        if op_type in "conv2d" and self._conv_op_index[
-                                op_index] < self._conv_op_index[index + 1]:
-                            continue
-                        elif op_type in Calibrator.non_conv_int8_op_type:
-                            # dequantize_op_index.append(op_index) #disable dequantize if enable force_fp32_output
-                            program.current_block().ops[op_index - 1]._set_attr(
-                                "force_fp32_output", True)
-                            break
-                        else:
-                            program.current_block().ops[op_index - 1]._set_attr(
-                                "force_fp32_output", True)
+            elif self._conv_op_index[index] + 1 < self._conv_op_index[index +
+                                                                      1]:
+
+                program.current_block().ops[self._conv_op_index[
+                    index]]._set_attr("force_fp32_output", True)
+
+                for op_index in range(self._conv_op_index[index + 1],
+                                      self._conv_op_index[index], -1):
+                    op_type = program.current_block().ops[op_index].type
+                    op_has_int8_input = False
+                    input_var_name = None
+                    input_length = len(program.current_block().ops[op_index]
+                                       .input_arg_names)
+
+                    for var_name in program.current_block().ops[
+                            op_index].input_arg_names:
+                        if self.__check_var_source_dt(var_name) != -1:
+                            op_has_int8_input = True
+                            input_var_name = var_name
                             break
 
-                    for op_index in range(self._conv_op_index[index + 1],
-                                          self._conv_op_index[index], -1):
-                        op_type = program.current_block().ops[op_index].type
-                        op_has_int8_input = False
-                        input_length = len(program.current_block().ops[op_index]
-                                           .input_names)
-                        for input_attr_name in program.current_block().ops[
-                                op_index].input_names:
-                            if program.current_block().ops[op_index].input(
-                                    input_attr_name):
+                    if op_has_int8_input:
+                        if op_type == "conv2d":
+                            if program.current_block().ops[op_index +
+                                                           1].type == "conv2d":
+                                continue
+                            elif program.current_block(
+                            ).ops[op_index +
```

Comment:
In the following cases
`...->conv2d->pool2d->conv2d->...`
`...->conv2d->pool2d->pool2d->conv2d->...`
(from the Detect model) the dequantize op will be inserted after the first `pool2d` op, but it should not. Am I right?

Reply:
yes, you're right. We're not considered such case. According to ur example, the ->conv2d->pool2d->pool2d->conv2d->.. will be ->conv2d(force_fp32_output)->pool2d->pool2d->quantize_op-->conv2d->... . It would low the int8 performance but w/o accuracy drop.
As we don't have plan to support detect model(BTW, would u please share me the model name of this detect model) for coming PaddlePaddle 1.3 release, do u think it's necessary to fix this issue for this PR?

Reply:
@wojtuss Thansk for your reporting. INT8 v1 targets the pre-defined POR models, including ResNet-50, MobileNet, SSD-MobileNet, and Se-ResNeXT. INT8 v2 will have the new design and will enhance the graph transformation with the pattern you mentioned.

Reply:
OK, please add some comments about the INT8 v1 targets. At least to the PR's description.

Reply:
Ok. Will add the comments on the class Calibrator doc.
Thank you.

Reply:
Updated.